### ISSUE TITLE
Rename does not rename methods across classes in hierarchy

### ISSUE THREAD

**Issue by NorbertHauriel** [microsoft/pylance-release#813]
Issue Type: <b>Bug</b>

If I refactor rename an overridden method, it does not rename the base class method, and vice versa. Also, no references are found. Basically, intellisense has no knowledge of polymorhpism, which can cause havoc in a large codebase, if someone does not know about this lack of feature.

I will provide additional info if needed.

Extension version: 2020.12.2
VS Code version: Code 1.52.1 (ea3859d4ba2f3e577a159bc91e3074c5d85c0523, 2020-12-16T16:34:46.910Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz (4 x 3200)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.94GB (1.25GB free)|
|Process Argv|--crash-reporter-id 504acee2-a3ea-41ce-a1e2-99e1630d8500|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
openlogontheside:30221877
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30211401
wsl2prompt:30224612
pythonvsdeb440:30237827
unusedpromptcf:30224611
folderexplorercf:30224615
openfilemenu:30224647
pythonvsded773cf:30236630
pythonvspyt859:30241729

```

</details>

**Comment by norbertHauriel** [comment]
Furthermore, docstring of base method is not inherited in child method, even when there is no child method docstring.

**Comment by jakebailey** [comment]
The original issue was about renaming class methods within a hierarchy, which requires some discussion/thought about what we think is the "correct" behavior; we currently have the safer behavior of not trying to modify a bunch of classes due to a rename.

That docstring example is a different issue, but I'd be curious to see the example because that would indicate there's a bug in the inheritance code as a whole (as we definitely find docstrings). Do you have a codebase that reproduces this?

**Comment by norbertHauriel** [comment]
You renamed my issue title to try to reduce the problem domain, but here I show you a demonstration where

- method renaming
- docstring inheritance
- reference finding

Don't work on my side. In the attached gif, I don't show the renaming, because that is not the root cause of this problem, just a symptom of not finding references.

![VS Code Pylance bug demonstration](https://user-images.githubusercontent.com/5183862/104279210-34169200-54aa-11eb-9818-d89cd5507d67.gif)

**Comment by jakebailey** [comment]
Hm, I guess I don't see that behavior as "wrong" per se; there are no uses of the child method; finding references to the child method shouldn't be going back to the base class the base class technically doesn't call the child class. Would have to see how this is done in other LSs.

RE: docs, I'd be interested in seeing what `help` says about each of these items in the child class; does Python's own builtin functions go to the base class for overriden functions? From it's POV, I'd have thought it would hit the function and say "no doc" because it can't really know that it's an override without walking the entire MRO to find one... but I know we can choose to do something different in the editor.

**Comment by norbertHauriel** [comment]
Local refactoring works very well as I've tried it, hoping that in the future, more advanced refactoring options will be possible in Pylance. Thank you for what we already have!

**Comment by heejaechang** [comment]
This issue has been fixed in version 2022.4.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202241-13-April-2022)
